### ISSUE TITLE
Line fitting models fail if the bin_centers vector is of type float32

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#193]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/193) by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

This will get in the way of full adoption of LMfit, so we want to fix it.

Here is a demo of simulated Mn K-alpha data being fit first with a float64, then a float32 vector of energies.

```python
import mass
import numpy as np
import pylab as plt

line = mass.spectra["MnKAlpha"]
model = line.model()
N = 100000
energies = line.rvs(size=N, instrument_gaussian_fwhm=2.2)  # draw from the distribution

for dtype in (np.float64, np.float32):
    e = np.asarray(energies, dtype=dtype)
    sim, bin_edges = np.histogram(e, 120, [5865, 5925])
    binsize = bin_edges[1] - bin_edges[0]
    e = bin_edges[:-1] + 0.5*binsize
    params = model.guess(sim, bin_centers=e)
    result = model.fit(sim, params, bin_centers=e)
    result.plot()
    print(result.fit_report())
```

Notice how the float32 looks bad and includes this text:

```
##  Warning: uncertainties could not be estimated:
    peak_ph:     at initial value
    dph_de:      at initial value
```

‌

The problem has a few steps:

1. `c, b = np.histogram(e,...)` returns bin edges `b` of type float64 for when `e` is either float64 _or any integer type_, but if `e` is of type float32, then `b` will be, too.
2. Then the MASS line model has lines of the form `energy = (bin_centers - peak_ph) / dph_de + self.spect.peak_energy` which subtract the float64 parameter `peak_ph` from the \(potentially\) float32 array `bin_centers`.

So the fix needs to be to do an `bin_centers = np.asarray(bin_centers, dtype=np.float)` at the right place. Plus tests, of course.

‌

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

In `GenericLineModel` convert bin_centers to float64

Fixes #193.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Removing milestone: v0.7 (automated comment)
